broker/pxc/cluster.go

```diff
@@ -0,0 +1,93 @@
+package pxc
+
+import (
+	"log"
+	"time"
+
+	"github.com/Percona-Lab/percona-dbaas-cli/broker"
+	"github.com/Percona-Lab/percona-dbaas-cli/dbaas"
+	"github.com/Percona-Lab/percona-dbaas-cli/dbaas/pxc"
+)
+
+const (
+	defaultVersion = "default"
+
+	noS3backupWarn = `[Error] S3 backup storage options doesn't set: %v. You have specify S3 storage in order to make backups.
+You can skip this step by using --s3-skip-storage flag add the storage later with the "add-storage" command.
+`
+)
+
+func (p *Controller) DeployPXCCluster(params ProvisionParameters, skipS3Storage *bool, instanceID string) error {
+	app := NewPXC(params.ClusterName, defaultVersion)
+	conf := Config{}
+	conf.SetDefault()
```

Comment:
It's a taste issue and I think not some important, but maybe it will be better to use a factory function like `NewDefaultConfig()` or just `NewConfig()` to have a predefined config in one single call? 

```go
conf := NewDefaultConfig()
```

```go
// NewDefaultConfig factory method that creates new config with predefined default parameters.
func NewDefaultConfig() *Config {
    return &Config{
	StorageSize: "6G",
	Instances: int32(3),
	RequestCPU: "600m",
	RequestMem: "1G",
	AntiAffinityKey: "kubernetes.io/hostname",
	ProxyInstances: int32(1),
	ProxyRequestCPU: "600m",
	ProxyRequestMem: "1G",
	ProxyAntiAffinityKey: "kubernetes.io/hostname",
	S3SkipStorage: true,
    }
}
```

It seems to have a method like `SetDefault()` make sense only if we would like to reset some config to defaults later.